runtime/protocol/http-client-engines/http-client-engine-crt/jvm/src/aws/smithy/kotlin/runtime/http/engine/crt/ConnectionManager.kt

```diff
@@ -61,16 +66,22 @@ internal class ConnectionManager(
         val manager = getManagerForUri(request.uri, proxyConfig)
         var leaseAcquired = false
 
+        metrics.queuedRequests = pending.incrementAndGet()
```

Comment:
**correctness**: If we are trying to acquire a connection then it's not queued right? I think queued would be before we hit the [requestLimiter](https://github.com/awslabs/smithy-kotlin/blob/2174a835e5bf088418db88368d4f888204ed9008/runtime/protocol/http-client-engines/http-client-engine-crt/jvm/src/aws/smithy/kotlin/runtime/http/engine/crt/CrtHttpEngine.kt#L63).

```kotlin
metrics.queuedRequests = pending.incrementAndGet()
requestLimiter.withPermit {
    metrics.queuedRequests = pending.decrementAndGet()
}

```

Reply:
OK, if "queued" means "before we attempt to acquire a connection" then I'm guessing that the `requestsQueuedDuration` measurement below is _also_ wrong. I'll move it too.

Reply:
Had to look at the definitions I gave them again and I think this would be inline with what it says

> queued=waiting to be executed (e.g. waiting for thread to be available), in-flight=actively processing

If we are to the point of acquiring a connection we are "actively processing" the request. I can see where the definition could be interpreted differently though as in "I have all the resources needed at this point to execute the request" but I think establishing a connection (or waiting on one to be available) is part of overall request processing. Curious if others disagree.

Reply:
This still doesn't look correct, I don't think `queuedRequests` is calculated in connection manager.

Reply:
Ah I think I was confusing the semaphore inside `ConnectionManager` with the one inside `CrtHttpEngine`. Switching.